### PR TITLE
Setup microsite

### PR DESCRIPTION
**Pull request by armanbilge**
After merging you will also need to configure your repository settings:

1. Grant "Read and write" permissions to workflows. This enables them to push to the gh-pages branch. https://github.com/typelevel/shapeless-3/settings/actions
2. Set the GitHub pages source to the / (root) directory on the gh-pages branch. https://github.com/typelevel/shapeless-3/settings/pages

### CODE REVIEW

**Comment by armanbilge** [comment]
The microsite is already equipped with mdoc:
https://scalameta.org/mdoc/

To start a live-preview server locally run `docs/tlSitePreview` in sbt.

File: build.sbt [hunk 1]
```diff
@@ -124,3 +124,5 @@ ThisBuild / developers := List(
   Developer("joroKr21", "Georgi Krastev", "joro.kr.21@gmail.com", url("https://twitter.com/Joro_Kr")),
   Developer("TimWSpence", "Tim Spence", "timothywspence@gmail.com", url("https://twitter.com/timwspence"))
 )
+
+lazy val docs = project.in(file("site")).enablePlugins(TypelevelSitePlugin)
```
Comment (joroKr21) [inline build.sbt:128]: Why `in(file("site"))`?

Comment (armanbilge) [reply build.sbt:128]: It can be anything you want except `docs/`, per mdoc instructions.
https://scalameta.org/mdoc/docs/installation.html#sbt

Comment (armanbilge) [reply build.sbt:128]: Note that the docs do still live in the `docs/` folder.

**Review by joroKr21 — APPROVED** [review]
Thanks, I will look into the settings later 👍

**Pull request merged**